### DISCUSSION TITLE
remote for custom bin codes still not working

### DISCUSSION THREAD

**Discussion by sp222222** [GerryDazoo/Slinger#23] [General]
IDK Gerry, I tried the latest version 3.07 and I still cannot get the basic power, numbers etc. In fact, i went backwards to yesterday setup and if I hadn't done the custom remote myself... now the "Advanced remote control buttons" I setup don't work anymore. Here's the ch46 and maybe power? 

xiting Remote Control Stream Handler for ('192.168.1.86', 58108)
solo Master Bedrm Got Streamer Control Message IR b'\x0e\x00\x00\x02'
solo Master Bedrm Sending IR command b'\x0e\x00\x00\x02'
.08/15/2022, 16:37:04   RemoteControl connection from ('192.168.1.86', 58113)

Starting remote control stream handler for  ('192.168.1.86', 58113) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('Signal Strength', '46'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:04] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 58113)
solo Master Bedrm Got Streamer Control Message IR b'.\x00\x00\x02'
solo Master Bedrm Sending IR command b'.\x00\x00\x02'
.08/15/2022, 16:37:17   RemoteControl connection from ('192.168.1.86', 58118)

Starting remote control stream handler for  ('192.168.1.86', 58118) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('Signal Strength', '46'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:17] "POST /Remote HTTP/1.1" 200 -
solo Master Bedrm Got Streamer Control Message IR b'.\x00\x00\x02'
solo Master Bedrm Sending IR command b'.\x00\x00\x02'
Exiting Remote Control Stream Handler for ('192.168.1.86', 58118)
.08/15/2022, 16:37:21   RemoteControl connection from ('192.168.1.86', 58125)

Starting remote control stream handler for  ('192.168.1.86', 58125) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('TIMER Programming (MAIN BUTTON)', '63'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:21] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 58125)
solo Master Bedrm Got Streamer Control Message IR b'?\x00\x00\x02'
solo Master Bedrm Sending IR command b'?\x00\x00\x02'

Starting remote control stream handler for  ('192.168.1.86', 51432) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('6', '14 : round'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:46:25] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 51432)
solo Master Bedrm Got Streamer Control Message IR b'\x0e\x00\x00\x02'
solo Master Bedrm Sending IR command b'\x0e\x00\x00\x02'

**Comment by sp222222** [comment]
got this with 3.04
Button Clicked ImmutableMultiDict([('6', '14 : round'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:55:03] "POST /Remote HTTP/1.1" 200 -
solo Master Bedrm Got Streamer Control Message IR b'\x0e\x00\x00\x00'
solo Master Bedrm Sending IR command b'\x0e\x00\x00\x00'
cmd: 0x87 err: 0x6 172
Exception in thread Thread-2 (streamer):
Traceback (most recent call last):
  File "threading.py", line 1016, in _bootstrap_inner
  File "threading.py", line 953, in run
  File "slingbox_server.py", line 655, in streamer
    name = slinginfo.get('name', '').strip()
  File "slingbox_server.py", line 266, in sling_cmd
    out = out + "%06d " % (cnt,) + ss + '\r\n'
  File "slingbox_server.py", line 69, in Decrypt
    chunk = [info[i], info[i+1]]
IndexError: list index out of range

frustrating esp since I actually got the "harder" buttons to start working once I went down the list of possibles.

**Comment by GerryDazoo** [comment]
This is the bug I was mentioning the other day. The Solo isn’t very happy when it get’s a IR message for an unconfigured input. You’ll see that the new version is using the proper VideoSource/Remote code. 0x02 instead of /x00 that the logs report here.

The actual error here is the slingbox sent back an error message that the server code couldn’t decode.

**Comment by GerryDazoo** [comment]
Well, the good news is the server is sending the correct remote code /x02 and the Solo isn’t complaining. The logs sure look like it’s doing the right thing. The fact that you can get some buttons working implies that the Solo is getting the message and doing “something”. Do you have an IR receiver laying about, a Flirc or something similar to verify that the Solo is sending something out when you click a button. 

My solo is configured with a private bin remote definition as well and it’s working as expected.

I’m assuming that you’ve verified everything is working OK using Slingplayer.  

All I can suggest is we get a packet capture of what the Slingplayer software is sending to your solo and make sure it’s the same thing my code is sending. We might need to do a zoom so I can walk you through it.

One more thing, On the old version of the code when you used the RT button to discover the right remote code do you recall what the sling box responded to? 

 

*	Gerry

 

From: sp222222 ***@***.***> 
Sent: Monday, August 15, 2022 5:51 PM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] remote for custom bin codes still not working (Discussion #23)

 

IDK Gerry, I tried the latest version 3.07 and I still cannot get the basic power, numbers etc. In fact, i went backwards to yesterday setup and if I hadn't done the custom remote myself... now the "Advanced remote control buttons" I setup don't work anymore. Here's the ch46 and maybe power?

xiting Remote Control Stream Handler for ('192.168.1.86', 58108)
solo Master Bedrm Got Streamer Control Message IR b'\x0e\x00\x00\x02'
solo Master Bedrm Sending IR command b'\x0e\x00\x00\x02'
.08/15/2022, 16:37:04 RemoteControl connection from ('192.168.1.86', 58113)

Starting remote control stream handler for ('192.168.1.86', 58113) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('Signal Strength', '46'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:04] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 58113)
solo Master Bedrm Got Streamer Control Message IR b'.\x00\x00\x02'
solo Master Bedrm Sending IR command b'.\x00\x00\x02'
.08/15/2022, 16:37:17 RemoteControl connection from ('192.168.1.86', 58118)

Starting remote control stream handler for ('192.168.1.86', 58118) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('Signal Strength', '46'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:17] "POST /Remote HTTP/1.1" 200 -
solo Master Bedrm Got Streamer Control Message IR b'.\x00\x00\x02'
solo Master Bedrm Sending IR command b'.\x00\x00\x02'
Exiting Remote Control Stream Handler for ('192.168.1.86', 58118)
.08/15/2022, 16:37:21 RemoteControl connection from ('192.168.1.86', 58125)

Starting remote control stream handler for ('192.168.1.86', 58125) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('TIMER Programming (MAIN BUTTON)', '63'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:21] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 58125)
solo Master Bedrm Got Streamer Control Message IR b'?\x00\x00\x02'
solo Master Bedrm Sending IR command b'?\x00\x00\x02'

Starting remote control stream handler for ('192.168.1.86', 51432) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('6', '14 : round'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:46:25] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 51432)
solo Master Bedrm Got Streamer Control Message IR b'\x0e\x00\x00\x02'
solo Master Bedrm Sending IR command b'\x0e\x00\x00\x02'

**Comment by GerryDazoo** [comment]
Let me noodle on this some more. I might need to send you a test version.

 

From: Gerry Dubois ***@***.***> 
Sent: Monday, August 15, 2022 6:51 PM
To: 'GerryDazoo/Slinger' ***@***.***>; 'GerryDazoo/Slinger' ***@***.***>
Cc: 'Subscribed' ***@***.***>
Subject: RE: [GerryDazoo/Slinger] remote for custom bin codes still not working (Discussion #23)

 

Well, the good news is the server is sending the correct remote code /x02 and the Solo isn’t complaining. The logs sure look like it’s doing the right thing. The fact that you can get some buttons working implies that the Solo is getting the message and doing “something”. Do you have an IR receiver laying about, a Flirc or something similar to verify that the Solo is sending something out when you click a button. 

My solo is configured with a private bin remote definition as well and it’s working as expected.

I’m assuming that you’ve verified everything is working OK using Slingplayer.  

All I can suggest is we get a packet capture of what the Slingplayer software is sending to your solo and make sure it’s the same thing my code is sending. We might need to do a zoom so I can walk you through it.

One more thing, On the old version of the code when you used the RT button to discover the right remote code do you recall what the sling box responded to? 

 

*	Gerry

 

From: sp222222 ***@***.*** ***@***.***> > 
Sent: Monday, August 15, 2022 5:51 PM
To: GerryDazoo/Slinger ***@***.*** ***@***.***> >
Cc: Subscribed ***@***.*** ***@***.***> >
Subject: [GerryDazoo/Slinger] remote for custom bin codes still not working (Discussion #23)

 

IDK Gerry, I tried the latest version 3.07 and I still cannot get the basic power, numbers etc. In fact, i went backwards to yesterday setup and if I hadn't done the custom remote myself... now the "Advanced remote control buttons" I setup don't work anymore. Here's the ch46 and maybe power?

xiting Remote Control Stream Handler for ('192.168.1.86', 58108)
solo Master Bedrm Got Streamer Control Message IR b'\x0e\x00\x00\x02'
solo Master Bedrm Sending IR command b'\x0e\x00\x00\x02'
.08/15/2022, 16:37:04 RemoteControl connection from ('192.168.1.86', 58113)

Starting remote control stream handler for ('192.168.1.86', 58113) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('Signal Strength', '46'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:04] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 58113)
solo Master Bedrm Got Streamer Control Message IR b'.\x00\x00\x02'
solo Master Bedrm Sending IR command b'.\x00\x00\x02'
.08/15/2022, 16:37:17 RemoteControl connection from ('192.168.1.86', 58118)

Starting remote control stream handler for ('192.168.1.86', 58118) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('Signal Strength', '46'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:17] "POST /Remote HTTP/1.1" 200 -
solo Master Bedrm Got Streamer Control Message IR b'.\x00\x00\x02'
solo Master Bedrm Sending IR command b'.\x00\x00\x02'
Exiting Remote Control Stream Handler for ('192.168.1.86', 58118)
.08/15/2022, 16:37:21 RemoteControl connection from ('192.168.1.86', 58125)

Starting remote control stream handler for ('192.168.1.86', 58125) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('TIMER Programming (MAIN BUTTON)', '63'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:37:21] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 58125)
solo Master Bedrm Got Streamer Control Message IR b'?\x00\x00\x02'
solo Master Bedrm Sending IR command b'?\x00\x00\x02'

Starting remote control stream handler for ('192.168.1.86', 51432) to port 8083
Remote Control Connected
Remote Code: 2 0x0 0x0 0x2
Button Clicked ImmutableMultiDict([('6', '14 : round'), ('Digits', '')])
127.0.0.1 - - [15/Aug/2022 16:46:25] "POST /Remote HTTP/1.1" 200 -
Exiting Remote Control Stream Handler for ('192.168.1.86', 51432)
solo Master Bedrm Got Streamer Control Message IR b'\x0e\x00\x00\x02'
solo Master Bedrm Sending IR command b'\x0e\x00\x00\x02'

**Comment by sp222222** [comment]
it responded to some keys in 60s and 40s range and I made the
remote page with those buttons and it worked but why it doesn’t work now is
really frustrating. Yes, all buttons on this master bedroom slingbox work
exactly as needed with slingplayer app, slingfront, websling etc.

I guess send a test code but I won’t be around this weekend to test it.
Maybe around this time during the week

Mike
